src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -37,11 +38,50 @@ protected override string GetDatabaseFilePath(string workingFolderPath)
         }
 
         protected override AbstractPersistentStorage OpenDatabase(Solution solution, string workingFolderPath, string databaseFilePath)
-            => new SQLitePersistentStorage(
-                OptionService, workingFolderPath, solution.FilePath, databaseFilePath, this.Release, _faultInjectorOpt);
+        {
+            // try to get db ownership lock. if someone else already has the lock. it will throw
+            var dbOwnershipLock = TryGetDatabaseOwnership(databaseFilePath);
+
+            return new SQLitePersistentStorage(
+                OptionService, workingFolderPath, solution.FilePath, databaseFilePath, this.Release, dbOwnershipLock, _faultInjectorOpt);
+        }
+
+        private static IDisposable TryGetDatabaseOwnership(string databaseFilePath)
+        {
+            try
+            {
+                // make sure directory exist first.
+                EnsureDirectory(databaseFilePath);
+
+                return File.Open(
+                    Path.Combine(Path.GetDirectoryName(databaseFilePath), LockFile),
+                    FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException("can't get the ownership", ex);
```

Comment:
I just followed existing pattern I think came from esent behavior where it would just throw Esent exception when it can't open the db. but sure I can change it to use return value.